# A/B Testing Homework 4


## Part 1: Compare to known click rate (Pa = .5)


In [3]:
from scipy.stats import norm 
import math
import matplotlib.pyplot as plt

### 1A) 

<img src='part1A.png' style='transform: rotate(270deg);'>

### 1b)

In [4]:
pa = .5
n = 50
Exbar = pa
Varxbar = (pa*(1-pa))/n
print("E(xbar)=", Exbar)
print("Var(xbar)=", Varxbar)


E(xbar)= 0.5
Var(xbar)= 0.005


### 1c)

In [5]:
pbhat = .6 
prob = 1 - norm.cdf(pbhat, Exbar, math.sqrt(Varxbar))
prob 

0.078649603525142719

### 1d) Should you reject the null hypothesis pb <= pa? Why? 

We shouldn't reject the null hypothesis because P(X > pb) is greater than the rejection level.

### 1e) 

In [6]:
n = 100
newVar = (pa*(1-pa))/n 
prob1 = 1 - norm.cdf(pbhat, Exbar, math.sqrt(newVar))
prob1

0.022750131948179209

We should reject the null hypothesis in this case because P(X > pb) is less than .05, the rejection level. 

### 1f)

In [7]:
qb = norm.ppf(.95, Exbar, math.sqrt(newVar))
print("The smallest phat value is,",qb)



The smallest phat value is, 0.582242681348


### 1g) 

In [8]:
qb1 = norm.ppf(.95, Exbar, math.sqrt(Varxbar))
print("The smallest detectable improvement is, ", qb1 - pa)

The smallest detectable improvement is,  0.116308715368


## Part 2: Compare to known click rate (pa = .75)

### 2a)

In [9]:
pa = .75
n = 50
Exbar = pa
Varxbar = (pa*(1-pa))/n
print("E(xbar)=", Exbar)
print("Var(xbar)=", Varxbar)

E(xbar)= 0.75
Var(xbar)= 0.00375


### 2b) 

In [10]:
pbhat = .6 
prob = 1 - norm.cdf(pbhat, Exbar, math.sqrt(Varxbar))
prob

0.99284706078228524

### 2c)

We shouldn't reject the null hypothesis because P(X > pb) is greater than the rejection rate of .05. Also we are only looking for an increase in click rate. 

### 2d) 

In [11]:
n = 100 
newVar = (pa*(1-pa))/n 
prob = 1 - norm.cdf(pbhat, Exbar, math.sqrt(newVar))
prob

0.99973399724743039

We shouldn't reject null hypothesis. 

### 2e) 

In [12]:
qb2 = norm.ppf(.95, pa, math.sqrt(newVar))
print("The smallest phat value is,",qb2)


The smallest phat value is, 0.821224251322


### 2f)

In [13]:
qb2 = norm.ppf(.95, pa, math.sqrt(Varxbar))
print("The smallest detectable improvement is,", qb2 - pa)

The smallest detectable improvement is, 0.10072630219


## Part 3: 

The smallest detectable improvement is larger in 1g rather then 2f. This could be because the variance is smaller in 2f (.00375) in comparison to the variance in 1g (variance = .005).

## Part 4: 

### 4a) 

<img src='part1B.png' style='transform: rotate(270deg);'>

### 4b) 

p = (sa + sb)/(na + nb)

### 4c)

In [14]:
sa = 35 
sb = 35 
na = 55 
nb = 45 
p = (sa + sb)/(na + nb)
Vary = (1/na + 1/nb)*(p*(1 - p))
Vary 

0.008484848484848486

### 4d)

In [15]:
yhat = (sb/nb) - (sa/na)
yhat

0.14141414141414144

### 4e) 

In [16]:
prob = 1 - norm.cdf(yhat, 0, math.sqrt(Vary))
prob

0.06236482228872664

### 4f) 

We shouldn't reject the null hypothesis because P(Y > y) is greater then .05. 

In [18]:
norm.cdf(norm.ppf(.5,loc=0, scale=1))

0.5